server/controllers/admin/allAdminProducts.ts

```diff
@@ -0,0 +1,16 @@
+import { Request, Response } from 'express';
+import { getProductsAdmin } from '../../queries';
+import { CustomError } from '../../utils';
+
+const getAllProductsAdmin = async (req:Request, res:Response) => {
+  try {
+    const data = await getProductsAdmin();
+    res.json({
+      data,
+    });
+    // res.json(Quantity);
+  } catch {
+    throw new CustomError(400, 'No Products');
+  }
```

Comment:
If an error happens in the query, then an error will be thrown here, so you need to pass it to the server middleware.